certification/internal/shell/operator_pkg_name_uniqueness_mounted.go

```diff
@@ -0,0 +1,233 @@
+// check.
+type OperatorPkgNameIsUniqueMountedCheck struct{}
+
+func (p *OperatorPkgNameIsUniqueMountedCheck) Validate(bundle string) (bool, error) {
+	annotations, err := p.getAnnotationsFromBundle(bundle)
+	if err != nil {
+		log.Errorf("unable to get annotations.yaml from the bundle")
+		return false, err
+	}
+
+	packageName, err := p.getPackageName(annotations)
+	if err != nil {
+		log.Errorf("unable to extract package name from ClusterServicVersion", err)
+		return false, err
+	}
+
+	log.Debugf("operator package name is %s", packageName)
+
+	req, err := p.buildRequest(apiEndpoint, packageName)
+	if err != nil {
+		log.Errorf("unable to build API request structure", err)
+		return false, err
+	}
+
+	resp, err := p.queryAPI(http.DefaultClient, req)
+	if err != nil {
+		log.Errorf("unable to query package name validation API for uniqueness check", err)
+		return false, err
+	}
+
+	data, err := p.parseAPIResponse(resp)
+	if err != nil {
+		log.Errorf("unable to parse response provided by package name validation API", err)
+		return false, err
+	}
+
+	return p.validate(data)
+}
+
+// getAnnotationsFromBundle will search a bundle's metadata directory for a file called annotations.yaml, which is
+// expected to exist in every bundle. This file's contents are read and marshaled to a simple map, which is then
+// returned.
+func (p *OperatorPkgNameIsUniqueMountedCheck) getAnnotationsFromBundle(mountedDir string) (map[string]string, error) {
+	log.Trace("reading annotations file from the bundle")
+	log.Debug("mounted directory is ", mountedDir)
+	annotationsFilePath := path.Join(mountedDir, "metadata", "annotations.yaml")
+
+	fileContents, err := ioutil.ReadFile(annotationsFilePath)
+	if err != nil {
+		log.Error("fail to read metadata/annotation.yaml file in bundle")
+		return nil, err
+	}
+
+	annotations, err := p.extractAnnotationsBytes(fileContents)
+	if err != nil {
+		log.Error("metadata/annotations.yaml found but is malformed")
+		return nil, err
+	}
+
+	return annotations, nil
+}
+
+// extractAnnotationsBytes reads the annotation data read from a file and returns the expected format for that yaml
+// represented as a map[string]string.
+func (p *OperatorPkgNameIsUniqueMountedCheck) extractAnnotationsBytes(annotationBytes []byte) (map[string]string, error) {
+	if len(annotationBytes) == 0 {
+		return nil, errors.New("the annotations file was empty")
+	}
+
+	var bundleMeta metadata
+	if err := yaml.Unmarshal(annotationBytes, &bundleMeta); err != nil {
+		log.Error("metadata/annotations.yaml found but is malformed")
+		return nil, err
+	}
+
+	return bundleMeta.Annotations, nil
+}
+
+// getPackageName accepts the annotations map and searches for the specified annotation corresponding
+// with the complete bundle name for an operator, which is then returned.
+func (p *OperatorPkgNameIsUniqueMountedCheck) getPackageName(annotations map[string]string) (string, error) {
+	key := "operators.operatorframework.io.bundle.package.v1"
```

Comment:
I would like to have this declared as a const on the top.

Reply:
@samira-barouti done!